src/components/JUPTable/JUPTable.tsx

```diff
@@ -0,0 +1,184 @@
+import { ITransaction } from "types/NXTAPI";
+import { visuallyHidden } from "@mui/utils";
+import { TableRowsPerPageOptions } from "utils/common/constants";
+
+// might still want to use this concept
+// function createWidgetRow(date: string, qty: number, toFrom: string): Data {
+//   return {
+//     date,
+//     qty,
+//     toFrom,
+//   };
+// }
+
+interface ITableTitleProps {
+  title: string;
+}
+const TableTitle: React.FC<ITableTitleProps> = ({ title }) => {
+  return (
+    <>
+      {
+        <TitleText variant="h6" id="tableTitle">
+          <SLink href="/transactions">{title}</SLink>
+        </TitleText>
+      }
+    </>
+  );
+};
+
+function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
+
+type Order = "asc" | "desc";
+
+function getComparator<Key extends keyof ITransaction>(
+  order: Order,
+  orderBy: Key
+): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
+  return order === "desc" ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
+}
+
+interface IEnhancedTableProps {
+  onRequestSort: (event: React.MouseEvent<unknown>, property: keyof Data) => void;
+  order: Order;
+  orderBy: string;
+  rowCount: number;
+  headCells: Array<IHeadCellProps>;
+}
+
+const EnhancedTableHead: React.FC<IEnhancedTableProps> = ({ onRequestSort, order, orderBy, rowCount, headCells }) => {
+  const createSortHandler = (property: keyof Data) => (event: React.MouseEvent<unknown>) => {
+    onRequestSort(event, property);
+  };
+
+  return (
+    <TableHead>
+      <TableRow>
+        {headCells.map((headCell) => (
+          <TableCell
+            key={headCell.id}
+            align={headCell.numeric ? "right" : "left"}
+            padding={headCell.disablePadding ? "none" : "normal"}
+            sortDirection={orderBy === headCell.id ? order : false}
+          >
+            <TableSortLabel
+              active={orderBy === headCell.id}
+              direction={orderBy === headCell.id ? order : "asc"}
+              onClick={createSortHandler(headCell.id)}
+            >
+              {headCell.label}
+              {orderBy === headCell.id ? (
+                <Box component="span" sx={visuallyHidden}>
+                  {order === "desc" ? "sorted descending" : "sorted ascending"}
+                </Box>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>
+        ))}
+      </TableRow>
+    </TableHead>
+  );
+};
+
+interface IJUPTableProps {
+  headCells: Array<IHeadCellProps>;
+  rows: Array<React.ReactElement>;
+  children?: Array<React.ReactElement>;
+  title: string;
+}
+
+const JUPTable: React.FC<IJUPTableProps> = ({ children, headCells, rows, title }) => {
+  const [rowsPerPage, setRowsPerPage] = React.useState(3);
```

Comment:
you may want to consider that for magic values like this, you pull them from the constants, especially when you define this in the constants. essentially you'd be saying 'use the first option of the rows allowed per page', so if that value ever updated you wouldn't be out of sync.

Reply:
agreed, done